numpy/core/tests/test_einsum.py

```diff
@@ -700,6 +700,12 @@ def test_einsum_fixed_collapsingbug(self):
         y2 = x[idx[:, None], idx[:, None], idx, idx]
         assert_equal(y1, y2)
 
+    def test_einsum_failed_on_p9_and_s390x(self):
+        # Issues gh-14692 and gh-12689
+        # Bug with signed vs unsigned char errored on power9 and s390x Linux
+        tensor = np.random.random_sample((10, 10, 10, 10))
+        print(np.einsum('ijij->', tensor))
+
```

Comment:
Maybe (please check locally before pushing)
```
x = np.einsum('ijij->', tensor)
y = tensor.trace(axis1=0, axis2=2).trace()
assert_equal(x, y)
```

Reply:
With floating point, you can't be sure that `x` and `y` will be *exactly* equal.  Use `assert_allclose` with an appropriately small tolerance, or use an integer array for `tensor` (assuming the test doesn't actually depend on the data type of the array).

Reply:
Done; tested on x86_64, power9, and s390x; and pushed.